_rules/image-button-non-empty-accessible-name-59796f.md

```diff
@@ -128,6 +130,14 @@ The image button has an empty `alt` attribute, and no other attributes that can
 
 #### Failed Example 3
 
+This image button has a non-empty `title` attribute, but an empty `alt` attribute. This gives the button an empty [accessible name][].
+
+```html
+<input type="image" src="/test-assets/shared/search-icon.svg" title="Search" alt="" />
```

Comment:
firefox says search button
chrome has no name (bug)
safari says submit button

firefox is correct here
chrome is wrong by providing no name
safari skips title but at least provides the fallback text.

Reply:
In my opinion this might be subjective (and that's probably also why browsers render this example in different ways).

From [4.3.1 input type="image" Accessible Name Computation](https://www.w3.org/TR/html-aam/#input-type-image-accessible-name-computation)

> 4.3.1 input type="image" Accessible Name Computation
    1. If the control has an [aria-label](https://www.w3.org/TR/wai-aria-1.2/#aria-label) or an [aria-labelledby](https://www.w3.org/TR/wai-aria-1.2/#aria-labelledby) attribute the [accessible name](https://www.w3.org/TR/accname-1.2/#dfn-accessible-name) is to be calculated using the algorithm defined in [Accessible Name and Description: Computation and API Mappings](https://www.w3.org/TR/accname-1.2/).
    2. Otherwise use alt attribute.
    3. Otherwise use title attribute.
    4. Otherwise if the previous steps do not yield a usable text string, the [accessible name](https://www.w3.org/TR/accname-1.2/#dfn-accessible-name) is a localized string of the word "Submit Query".
If none of the above yield a usable text string there is no [accessible name](https://www.w3.org/TR/accname-1.2/#dfn-accessible-name).

In our example the alt attribute has been clearly defined as empty (alt=""). Depending on interpretations, we might stop at point 2, considering the alt="" as a choice to make the image decorative, or we can move forward reviewing the title.

Even more important is that for images, the HTML Accessibility API Mapping contemplate (rightly) the specific scenario ([4.10 img Element](https://www.w3.org/TR/html-aam/#img-element-accessible-name-computation))

> 4.10.1 img Element Accessible Name Computation
    1. If the img element has an [aria-label](https://www.w3.org/TR/wai-aria-1.2/#aria-label) or an [aria-labelledby](https://www.w3.org/TR/wai-aria-1.2/#aria-labelledby) attribute the [accessible name](https://www.w3.org/TR/accname-1.2/#dfn-accessible-name) is to be calculated using the algorithm defined in [Accessible Name and Description: Computation and API Mappings](https://www.w3.org/TR/accname-1.2/).
    2. Otherwise use alt attribute, even if its value is the empty string.
    NOTE
    An img with an alt attribute whose value is the empty string is mapped to the [presentation](https://www.w3.org/TR/core-aam-1.2/#role-map-presentation) role. It has no accessible name.
    3. Otherwise, if there is no alt attribute use the title attribute.
    4. Otherwise there is no [accessible name](https://www.w3.org/TR/accname-1.2/#dfn-accessible-name).

Combining these two items, might result unclear which are expectations.

Interesting also the example provided by @Jym77 when images are not loaded.

Reply:
In addition, if these are the expectations, it is necessary that all browsers move in this direction; right now, this example might be an important barrier for some user depending on the technology the user is using.

Reply:
In the light of the recent clarifications, this should be a Passed Example indeed (since the empty `alt` does not provide a "usable string").
Description can say that the image button has name "Search", but some UA/AT (incorrectly) either use the empty `alt` or the fallback "Submit query".